cities/weather_service.py

```diff
@@ -0,0 +1,24 @@
+import os
+
+import httpx
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+async def fetch_weather(city_name: str):
+    api_key = os.getenv("WEATHER_API_KEY")
```

Comment:
Ensure that the environment variable `WEATHER_API_KEY` is correctly set in your environment. If it's not set, `api_key` will be `None`, which will cause the API request to fail.